src/mapsector.h

```diff
@@ -62,8 +62,14 @@ class MapSector
 	// Returns an owning ptr to block.
 	std::unique_ptr<MapBlock> detachBlock(MapBlock *block);
 
+	// This makes a copy of the internal collection.
+	// Prefer getBlocks() if possible.
 	void getBlocks(MapBlockVect &dest);
 
+	// Get access to the internal map. Only allowed from a const MapSector.
+	const auto & getBlocks() const { return m_blocks; }
+	const auto & getBlocks() = delete;
```

Comment:
```suggestion
	// Get access to the internal collection
	// This is explicitly only allowed on a const object since modifying anything while iterating is unsafe.
	// The caller needs to make sure that this does not happen.
	const auto &getBlocks() const { return m_blocks; }
	const auto &getBlocks() = delete;
```

Reply:
also I wasn't sure here if an `auto` type is allowed on a deleted function, it doesn't really have a return type...
But we can use it if no compiler complains